server/rosetta/codec.go

```diff
@@ -0,0 +1,22 @@
+package rosetta
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	authcodec "github.com/cosmos/cosmos-sdk/x/auth/types"
+	bankcodec "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// MakeCodec generates the codec required to interact
+// with the cosmos APIs used by the rosetta gateway
+func MakeCodec() (*codec.ProtoCodec, codectypes.InterfaceRegistry) {
```

Comment:
why do you need to use this function? isn't the one defined by the app enough?

Reply:
It exists(ed) to support rosetta instances running decoupled from application code itself, in fact it contained a minimum viable codec to parse the transactions in any arbitrary chain.